### PR TITLE
Added a login and logout system.

### PR DESCRIPTION
**Pull request by te4-gabriel-stark**
# Pull Request Template

## What issue did you implement or fix?
Fixed issues #2, #10 and #24 

## Summary
- Added a login system and changed a little bit in the backend.
- Added a logout button.

## Type of change
- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

## How can this implementation be improved?
- The backend isn't completed yet and vuetify isn't implemented yet.

### CODE REVIEW

File: frontend/src/store/modules/UserModule.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+import store from '@/store';
+import jwt from 'jsonwebtoken';
+import axios from 'axios';
+
+export interface ILoginPayload {
+    email: string,
+    password: string
+}
+
+interface IDecodedPayload {
```
Comment (TE4-Daniel-Kull) [inline frontend/src/store/modules/UserModule.ts]: Tillför detta interfacet något till projektet? Kommer vi någonsin att utöka den?

File: backend-log-o-meme/todo_backend.rb [hunk 2]
```diff
@@ -113,7 +113,7 @@ def authenticated?
 
     user_form = JSON.parse(request.body.read)
 
-    user = @db.execute('SELECT * FROM users WHERE username = ?', user_form['username']).first
+    user = @db.execute('SELECT * FROM users WHERE username = ?', user_form['email']).first
```
[the remark this answers is not in the document]
Comment (te4-gabriel-stark) [reply backend-log-o-meme/todo_backend.rb]: Eftersom att vi ska använda Email senare när det inte är denna skit versionen så skall ändrade jag vad jag skickar in, vilket gjorde att user_form['username'] inte fanns.

**Comment by TE4-emanuel-strom** [comment]
+1

File: frontend/src/components/LoginFormComponent.vue [hunk 3]
```diff
@@ -0,0 +1,48 @@
+<template>
+    <div id="LoginForm">
+        <form method="post" v-if="!UserModule.currentUser">
+            <input type="email" v-model="email" placeholder="Email">
+            <input type="password" v-model="password" placeholder="Password">
+            <button type="submit" @click="login">Login</button>
+        </form>
+        <button @click="logout" v-else>Logout</button>
```
[the remark this answers is not in the document]
Comment (te4-gabriel-stark) [reply frontend/src/components/LoginFormComponent.vue]: Log ut knappen ska senare till navbar och nu kan man säga UserModule.currentUser.username och få namnet

**Pull request merged**